presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeUtils.java

```diff
@@ -295,14 +321,15 @@ else if (type == ConnectorSession.class) {
         }
         block.append(invoke(binding, name));
 
-        if (function.isNullable()) {
+        if (bestChoice.getReturnPlaceConvention() == ReturnPlaceConvention.STACK && function.isNullable()) {
```

Comment:
Turn this into `switch`. And add assertion in the `PROVIDED_BLOCK` branch.

I recommend placing `function.isNullable()` outside the `switch`.

Please go through all comparisons involving `ReturnPlaceConvention` literals. In general, I recommend using `switch` instead of `if`s.